File: tests/basic/variable_bind.cpp
typedef struct {
  int x[5];
} int_5;


int_5 test () {
  return {{0,1,2,3,4}};
}

void pointer_arg(int* a, int b){
  *a =b ;
}


int main () {

  int arr[5] = {0,1,2,3,4};
  int x;
  int y;
  pointer_arg(&x, y);
  return 0;
}

File: tests/variable/bind/variable_bind.ml
open Optitrust
open Target

let _ = Run.script_cpp (fun _ ->
  !! Variable_basic.bind "a" ~const:true [cFunDef "test"; cReturn; cArrayInit];
  !! Variable_basic.bind "b" [cVarDef "arr"; cArrayInit];

  (* FIXME:
  !! Variable_basic.bind ~const:true "a0" [nbMulti; sExpr "arr[0]"];
     ~===~ [cCellAccess ()]
  *) (* LATER: remove the show *)
  (*show [cCellAccess ~base:[cVar "arr"] ()];
  show [cAccesses ~base:[cStrict; cCellAccess ~base:[cVar "arr"] ()] ()];*)
  !! Variable_basic.bind "a0" [nbMulti; cRead ~addr:[cAccesses ~base:[cStrict; cCellAccess ~base:[cVar "arr"] ()] ()] ()];

  !! Variable_basic.bind "z" [cFun "pointer_arg"; dArg 0];
)
